### DISCUSSION TITLE
Copilot improvements

### DISCUSSION THREAD

**Discussion by gldtn** [zed-industries/zed#7211] [Feature Requests]
Is it possible to currently chat with Copilot? Could not find a way to. I'm also missing the following from copilot..

- Chat window
- Start inline chat
- Explain this
- Fix this
- Generate Docs
- Generate Test

**Comment by JosephTLyons** [comment]
Right now, we only have copilot suggestions and then our [assistant](https://zed.dev/docs/assistant-panel), which is similar to Copilot chat, but requires an OpenAI key.

**Reply by Frank-vdm** [reply]
It's only kinda there, copilot chat has more contextual knowledge of the code base as opposed to the current ai assistant panel implementation (unless I'm missing something)

**Reply by Henvag** [reply]
In the future you should consider having the option for both instead of just the OpenAI key.

**Reply by y0sher** [reply]
@JosephTLyons The inline chat is one of the features I use the most in copilot for vscode. Being able to get suggestions without leaving the code is priceless. Does this just requires implementation or its also missing some support from github? It seems like the vs code ext is closed source..

**Comment by oir** [comment]
Is it possible to partially accept the copilot suggested autocompletion? In VSCode I often use cmd+right arrow (instead of tab) to word by word grab a correct prefix instead of the entire suggestion.

**Reply by JosephTLyons** [reply]
I didn't even know this was possible - that sounds like a neat addition!  We don't have that yet, but it would be really cool.

**Comment by JosephTLyons** [comment]
As of version [0.147.0-pre](https://zed.dev/releases/preview/0.147.0), we now have support for Copilot chat in the assistant. This will go out to stable tomorrow.
